doc/releases/migration-guide-3.6.rst

```diff
@@ -157,6 +157,11 @@ Networking
   a NULL pointer. IGMPv3 can be enabled using the Kconfig ``CONFIG_NET_IPV4_IGMPV3``.
   (:github:`65293`)
 
+* The network stack uses now separate IPv4 TTL (time-to-live) value for multicast packets. Before
+  the same TTL value was used for unicast and multicast packets. The IPv6 hop limit value is also
+  changed so that unicast and multicast packets can have different hop limit values.
+  (:github:`65886`)
```

Comment:
Some grammar fixes below, but wondering if this is not "just" something for the release notes? Are people's existing apps expected to *break* as per the described change, or is it just that from now on people have more flexibility w.r.t. configuring TTLs?

It might also make sense to mention the new Kconfig options (but again, might be a "release note" thing rather than migration guide, idk)

```suggestion
* The network stack now uses separate IPv4 TTL (time-to-live) values for multicast packets. Before,
  the same TTL value was used for unicast and multicast packets. The IPv6 hop limit value is also
  changed so that unicast and multicast packets can have different hop limit values.
  (:github:`65886`)
```

Reply:
> > The network stack now uses separate IPv4 TTL (time-to-live) values for multicast packets. Before,
> 
> @kartben Hmm, about the plural "values" that sounds wrong as there is just one "value". Each application can have their own value so perhaps "values" could be used but it kind of feels wrong. WDYT?

ah! you're right :) I fixed the wrong end of the sentence I guess, just updated the suggested change. and ACK re: waiting for others like @rlubos to chime in, you guys def know better in any case :)

Reply:
I slightly modified the IPv6 hop limit sentence, does this look ok now?
```
* The network stack now uses a separate IPv4 TTL (time-to-live) value for multicast packets.
  Before, the same TTL value was used for unicast and multicast packets.
  The IPv6 hop limit value is also changed so that unicast and multicast packets can have a
  different one. (:github:`65886`)
```